### ISSUE TITLE
[PRE REVIEW]: An Applied Geographic Information Systems and Science Course in R

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#85]
**Submitting author:** @andrewmaclachlan (<a href="http://orcid.org/0000-0003-0356-1623">Andrew MacLachlan</a>)
**Repository:** <a href="https://github.com/andrewmaclachlan/CASA0005repo" target ="_blank">https://github.com/andrewmaclachlan/CASA0005repo</a>
**Version:** 2.0
**Editor:** @ttimbers
**Reviewers:** tbeuzen

**Author instructions**

Thanks for submitting your paper to JOSE @andrewmaclachlan. **Currently, there isn't an JOSE editor assigned** to your paper.

@andrewmaclachlan if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/jose-reviewers) have already agreed to review for JOSE and may be suitable for this submission.

**Editor instructions**

The JOSE submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #85 with the following error: 

 Can't find any papers to compile :-(

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #85 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/85 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by whedon** [comment]
PDF failed to compile for issue #85 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/85 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-446a0298a33b/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by labarba** [comment]
@openjournals/dev Can you help us find out what is causing these errors? The [paper.md](https://github.com/andrewmaclachlan/CASA0005repo/blob/master/paper.md) file _is there_.

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/jose-papers/blob/jose.00085/jose.00085/10.21105.jose.00085.pdf)

**Comment by arfon** [comment]
I think Whedon is timing out here because of the very large size of this repository and therefore reporting it can't find the paper (apologies for the unhelpful error message).

@andrewmaclachlan - your repository is ~1.4Gb (> 3Gb on disk) which means Whedon is going to struggle with this repository (as will your users). Is there some way you could reduce the size of the repository?

**Comment by andrewmaclachlan** [comment]
Hi there, looks like Weedon did run (i assume it just creates the pdf proof) and that seems fine to me. I tried to stored some satellite imagery using GitHub LFS, however, it didn't work but means i have to delete the repo to remove them fully. I can do this if needed.

**Comment by andrewmaclachlan** [comment]
@weedon commands

**Comment by andrewmaclachlan** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by andrewmaclachlan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00085/jose.00085/10.21105.jose.00085.pdf)

**Comment by andrewmaclachlan** [comment]
Hi @labarba. I've managed to significantly reduce the size of the repo and correct some of the missing references in the article proof. I'd be happy for any of the following to review if they are able to:

Karthik Ram - based on their reproducibility work in R
Matthew Forshaw - based on similar disciplines and uses R
Joseph Stachelek - based on geographical area of work and uses R

Please let me know if you need anymore suggestions. The repo and website are being updated over this term to fix any issues with delivery, however all of the content will remain the same.

**Comment by andrewmaclachlan** [comment]
Hi @labarba and @arfon 

Just wondering if this has been accepted as i've found it here under accepted papers here: https://github.com/openjournals/jose-papers/blob/jose.00085/jose.00085/10.21105.jose.00085.pdf

Please let me know if i need to do anything.

Thanks

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00085/jose.00085/10.21105.jose.00085.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00085/jose.00085/10.21105.jose.00085.pdf) :page_facing_up: :point_left:

**Comment by labarba** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=1.20 s (108.7 files/s, 44275.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            18            597             31          18532
JavaScript                      36           1709           1737           9982
SVG                              5              0              0           3908
CSS                             23            747            174           3780
Rmd                             20           3517           4578           2876
TeX                              4              9              0            336
SQL                              5              5              0             87
Markdown                         2             38              0             75
R                                1             11             15             66
YAML                             2              0              0             56
XML                              8              0              0             23
JSON                             2              0              0             20
Bourne Shell                     2              4              3             15
Dockerfile                       2              7              0             12
-------------------------------------------------------------------------------
SUM:                           130           6644           6538          39768
-------------------------------------------------------------------------------


Statistical information for the repository '9270eb9252fb8e325dca3fb6' was
gathered on 2021/05/23.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
andrewmaclachlan                 5         15874           1486          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
andrewmaclachlan          14388           90.6          0.0               13.34
```

**Comment by labarba** [comment]
@darribas 👋  — You [published](https://jose.theoj.org/papers/10.21105/jose.00042) with JOSE a while back. Could I persuade you to help us out with a review? We have received this submission that I think is right up your alley: "An Applied Geographic Information Systems and Science Course in R" https://github.com/openjournals/jose-reviews/issues/85
Let me know…

**Comment by labarba** [comment]
@jsta 👋  — You [reviewed](https://jose.theoj.org/papers/10.21105/jose.00042) for JOSE a while back. Would you be able to contribute a review once more? We have a new submission that I think you could help us with: "An Applied Geographic Information Systems and Science Course in R" https://github.com/openjournals/jose-reviews/issues/85
Let me know…

**Comment by darribas** [comment]
@labarba thanks for thinking of me. Yes, this looks great and right up my alley, I'd love to be one of the reviewers. What're the next steps to follow on?

**Comment by labarba** [comment]
@whedon assign @ttimbers as editor

**Comment by whedon** [comment]
OK, the editor is @ttimbers

**Comment by labarba** [comment]
hi @darribas — thanks for volunteering! @ttimbers will edit this submission, and is looking to secure a second reviewer. Once that happens she'll start the review (which opens in another issue).

**Comment by jsta** [comment]
@labarba Apologies but I do not have availability to review currently.

**Comment by ttimbers** [comment]
Thanks for the reply @jsta . I will look for another reviewer.

**Comment by darribas** [comment]
Hello, just checking in to make sure nothing is waiting on me: has the review been started? If so, I think I should be able to get to it this month. Thanks! (cc @ttimbers )

**Comment by andrewmaclachlan** [comment]
Hi Dani and all

Thanks for agreeing to review this submission, just to make you aware we have changed the module for the forthcoming year. But each year we take a snapshot, which are listed on the previous version section of the main repository. The 2020-2021 module aligns with the submitted paper.

Let me know if that’s not clear

Thanks

Andy
________________________________

**Comment by labarba** [comment]
Looks like we're in need of a second reviewer here. Julia @jwagemann : might you be able to help us here, either with a review of this submission, or maybe pointing us to someone else who could contribute a review?

**Comment by labarba** [comment]
@andrewmaclachlan — You signed up to review for JOSE a while back. Would you be available to help us with this submission? "An Applied Geographic Information Systems and Science Course in R" https://github.com/openjournals/jose-reviews/issues/85
Let me know…

**Comment by andrewmaclachlan** [comment]
Hi @labarba, that's actually my submission so I can't be of much help.

**Comment by labarba** [comment]
Ha ha, that shows how much I'm struggling. I totally missed it—and this is the _second_ time that I make the mistake of pinging the author for a review! I guess at least my matching of interest and field of expertise is working ok? 🤷‍♀️ 
Try, try again!

**Comment by jgorzalski** [comment]
@whedon assign @darribas as reviewer

**Comment by whedon** [comment]
OK, @darribas is now a reviewer

**Comment by whedon** [comment]
OK, tbeuzen is now a reviewer

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/jose-reviews/issues/141.
